src/convertToObject.js

```diff
@@ -6,7 +6,21 @@
  * @return {object}
  */
 function convertToObject(sourceString) {
-  // write your code here
+  const obj = {};
+  const arr = sourceString
+    .split(';')
+    .map((row) => row.trim())
+    .filter(Boolean);
+
+  for (const row of arr) {
```

Comment:
Avoid using loops in this task. Instead, use iteration methods like 'forEach', 'map', or 'reduce' to process the array.